### PR TITLE
[GraphBolt] `gb.DataLoader` can simply be a datapipe.

### PR DESCRIPTION
**Pull request by mfbalin**
## Description
There is no need to derive from DataLoader as we are not using any of its functionalities.

I also remove some datapipe modification only meant for `num_workers>0` to a `if num_workers > 0:` branch.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/graphbolt/dataloader.py [hunk 1]
```diff
@@ -75,7 +75,7 @@ def __iter__(self):
         yield from self.dataloader
 
 
-class DataLoader(torch_data.DataLoader):
+class DataLoader(MiniBatchTransformer):
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/dataloader.py:78]: why not inherit from `IterDataPipe`?

File: tests/python/pytorch/graphbolt/test_integration.py [hunk 2]
```diff
@@ -64,19 +64,19 @@ def test_integration_link_prediction():
                         [3, 2],
                         [3, 2],
                         [3, 3],
-                        [5, 0],
-                        [5, 0],
-                        [3, 3],
+                        [5, 5],
```
Comment (Rhett-Ying) [inline tests/python/pytorch/graphbolt/test_integration.py:67]: why the expected results are changed?

File: python/dgl/graphbolt/dataloader.py (hunk 1 above)
Comment (mfbalin) [reply python/dgl/graphbolt/dataloader.py:78]: MiniBatchTransformer ensures returned objects are `gb.MiniBatch` objects.

File: tests/python/pytorch/graphbolt/test_integration.py (hunk 2 above)
Comment (mfbalin) [reply tests/python/pytorch/graphbolt/test_integration.py:67]: I guess torch.utils.data.DataLoader does things that we don't know about.

Comment (mfbalin) [reply tests/python/pytorch/graphbolt/test_integration.py:67]: The code is here, it is quite long so we need good justification to keep deriving from it. https://github.com/pytorch/pytorch/blob/main/torch/utils/data/dataloader.py

**Comment by mfbalin** [comment]
@frozenbugs this is a small PR that can potentially affect performance slightly. Do you want to take a quick look as well?

File: tests/python/pytorch/graphbolt/test_integration.py (hunk 2 above)
Comment (frozenbugs) [reply tests/python/pytorch/graphbolt/test_integration.py]: This is weird, it seems that the seed is respected in a different way.

**Comment by frozenbugs** [comment]
@dgl-bot

**Review by frozenbugs — APPROVED** [review]
